### PR TITLE
Added test case for add capacity flow to ceph for cluster health status 

### PR DESCRIPTION
**Pull request by a2batic**
Merge after:https://github.com/openshift/console/pull/3609

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+let clusterJSON;
+let previousCnt;
+let previousPods;
+let updatedClusterJSON;
+let updatedCnt;
+let updatedPods;
+let clusterExist = true;
+
+describe('Check availability of cluster', () => {
```
Comment (umangachapagain) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts]: Ideally, `describe` block would have an `it` block or nested `describe`. Or, you could use `tests` that is basically a merge of `describe + it`.

If we have HEALTH_WARN, does it mean cluster is not available?

Also, be specific about which cluster we are referring to. We also have NooBaa and OCS StorageCluster.

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      /* since rook-ceph-osd-prepare-ocs-deviceset- keeps changing their last 4 characters,
+      hence subtracting the count of previous  rook-ceph-osd-prepare-ocs-deviceset- pods */
+      expect(newPods.length - previousCnt * 3).toEqual(6);
+    });
+
+    it('Ceph cluster should be healthy after expansion', () => {
+      const cephValueAfter = JSON.parse(
+        execSync(`kubectl get cephCluster -n ${namespace} -o json`).toString(),
+      );
+      expect(cephValueAfter.items[0].status.ceph.health).not.toBe('HEALTH_ERR');
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts]: Move `HEALTH_ERR` to /utils/consts.ts

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts (hunk 1 above)
Comment (gnehapk) [reply frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts]: As mentioned, please specify ocs cluster here and for rest other areas.

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts [hunk 3]
```diff
@@ -24,84 +24,105 @@ let previousPods;
...
+      );
+      expect(isAllPodRunning).toEqual(true);
+    });
 
-  it('No OCP/OCS nodes should go to NotReady state', () => {
-    const nodes = JSON.parse(execSync(`kubectl get nodes -o json`).toString());
-    const isAllNodes = nodes.items.every((node) => isNodeReady(node));
+    it('No OCP/OCS nodes should go to NotReady state', () => {
+      const nodes = JSON.parse(execSync(`kubectl get nodes -o json`).toString());
+      const isAllNodes = nodes.items.every((node) => isNodeReady(node));
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts]: run forEach and directly expect inside the loop.

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts [hunk 4]
```diff
@@ -24,84 +24,105 @@ let previousPods;
...
 
-  it('No ocs pods should get restarted unexpectedly', () => {
-    const isAllPodRunning = previousPods.items.every(
-      (pod) => pod.status.phase === 'Running' || pod.status.phase === 'Succeeded',
-    );
-    expect(isAllPodRunning).toEqual(true);
-  });
+    it('No ocs pods should get restarted unexpectedly', () => {
+      const isAllPodRunning = previousPods.items.every(
+        (pod) => pod.status.phase === 'Running' || pod.status.phase === 'Succeeded',
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts]: getPhase

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts [hunk 5]
```diff
@@ -24,84 +24,105 @@ let previousPods;
...
-    previousPods = JSON.parse(execSync(`kubectl get pods -n ${namespace} -o json`).toString());
+if (cephValue.items[0]) {
+  describe('Check add capacity functionality for ocs service', () => {
+    beforeAll(async () => {
+      const { name } = storageCluster.items[0].metadata;
+      clusterJSON = JSON.parse(
+        execSync(`kubectl get -o json -n ${namespace} ${kind} ${name}`).toString(),
+      );
+      previousCnt = _.get(clusterJSON, 'spec.storageDeviceSets[0].count', undefined);
+      const uid = _.get(clusterJSON, 'metadata.uid', undefined).toString();
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts]: undef not reqd

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts [hunk 6]
```diff
@@ -24,84 +24,105 @@ let previousPods;
...
+});
 
-    const { name } = storageCluster.items[0].metadata;
-    clusterJSON = JSON.parse(
-      execSync(`kubectl get -o json -n ${namespace} ${kind} ${name}`).toString(),
-    );
-    previousCnt = _.get(clusterJSON, 'spec.storageDeviceSets[0].count', undefined);
-    const uid = _.get(clusterJSON, 'metadata.uid', undefined).toString();
-    previousPods = JSON.parse(execSync(`kubectl get pods -n ${namespace} -o json`).toString());
+if (cephValue.items[0]) {
```
Comment (umangachapagain) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts]:
```
if (! cephValue.items[0]) {
    it('Should state that cluster doesnt exist', () => {
      expect(cephValue.items[0]).toBe(undefined);
    });
}
```
Should be enough, isn't it?

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts [hunk 7]
```diff
@@ -24,84 +24,105 @@ let previousPods;
...
+      );
+      previousCnt = _.get(clusterJSON, 'spec.storageDeviceSets[0].count', undefined);
+      const uid = _.get(clusterJSON, 'metadata.uid', undefined).toString();
+      previousPods = JSON.parse(execSync(`kubectl get pods -n ${namespace} -o json`).toString());
 
-    await browser.get(
-      `${appHost}/k8s/ns/openshift-storage/operators.coreos.com~v1alpha1~ClusterServiceVersion`,
-    );
+      await browser.get(
+        `${appHost}/k8s/ns/openshift-storage/operators.coreos.com~v1alpha1~ClusterServiceVersion`,
```
Comment (umangachapagain) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts]:
```suggestion
`${appHost}/k8s/ns/{NS}/operators.coreos.com~v1alpha1~ClusterServiceVersion`,
```

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts [hunk 8]
```diff
@@ -24,84 +24,105 @@ let previousPods;
...
 
-    updatedCnt = _.get(updatedClusterJSON, 'spec.storageDeviceSets[0].count', undefined);
-    const statusCol = storageClusterRow(uid).$('td:nth-child(4)');
+      updatedCnt = _.get(updatedClusterJSON, 'spec.storageDeviceSets[0].count', undefined);
+      const statusCol = storageClusterRow(uid).$('td:nth-child(4)');
 
-    // need to wait as cluster states fluctuates for some time. Waiting for 2 secs for the same
-    browser.sleep(2000);
+      // need to wait as cluster states fluctuates for some time. Waiting for 2 secs for the same
+      browser.sleep(2000);
```
Comment (umangachapagain) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts]:
```suggestion
browser.sleep(2 * SECOND);
```

File: frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts [hunk 9]
```diff
@@ -24,84 +24,105 @@ let previousPods;
...
-    );
+      await browser.wait(until.textToBePresentInElement(statusCol, 'Progressing'));
+      await browser.wait(
+        until.textToBePresentInElement(statusCol.$('span.co-icon-and-text span'), 'Ready'),
+      );
 
-    updatedPods = JSON.parse(execSync(`kubectl get pod -o json -n ${namespace}`).toString());
-  }, 150000);
+      updatedPods = JSON.parse(execSync(`kubectl get pod -o json -n ${namespace}`).toString());
+    }, 150000);
```
Comment (umangachapagain) [inline frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts]:
```suggestion
updatedPods = JSON.parse(execSync(`kubectl get pod -o json -n ${namespace}`).toString());
  }, 15 * MINUTE);
```

**Comment by gnehapk** [comment]
Closing this PR as the changes are sent in https://github.com/openshift/console/pull/3609

**Pull request closed without merging**